zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
```

Comment:
Another question for @CaptainKanuk and/or @twiecki:

- Should we expose other correlation measures? Various internet resources also mention [Spearman Rank Correlation](http://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.stats.spearmanr.html) and [Kendall's Tau Coefficient](http://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.stats.kendalltau.html#scipy.stats.kendalltau).